kernel: bump 5.4 to 5.4.86

graysky2 opened the pull request:
Ran `update_kernel.sh` in a fresh clone without any existing toolchains.

Removed upstreamed patches:
```
  generic/pending-5.4/499-mtd-parser-cmdline-Fix-parsing-of-part-names-with-co.patch
```

Manually merged:
  adschm to edit

Build system: x86_64
Build-tested: ipq806x/R7800, bcm27xx/bcm2711
Run-tested: ipq806x/R7800

No dmesg regressions, everything functional

Signed-off-by: John Audia <graysky@archlinux.us>

graysky2 commented:
I attempted to build layerscape/LS1043A-RDB Default on https://github.com/openwrt/openwrt/commit/93b612221208c731da9b6c7100af636894405f75 but the build failed on the bootloader.  Note this is BEFORE my PR.  Given the level of changes needed for the layerscape patches, someone should look carefully at them.

adschm commented on target/linux/generic/pending-5.4/611-netfilter_match_bypass_default_table.patch:
```diff
@@ -76,17 +76,17 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 -	local_bh_disable();
  	addend = xt_write_recseq_begin();
--	private = READ_ONCE(table->private); /* Address dependency. */
+-	private = rcu_access_pointer(table->private);
```
What's removed here is just added a few lines above in the same patch again. I assume (without understanding the code) that the code should also be changed according to https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable-rc.git/commit/net/ipv4/netfilter/ip_tables.c?h=queue/5.4&id=b17244cebb24b30b886d46938cc5f798f7337574 there ...

adschm commented on target/linux/generic/pending-5.4/611-netfilter_match_bypass_default_table.patch:
```diff
@@ -76,17 +76,17 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 -	acpar.hotdrop = false;
 -	acpar.state   = state;
- 
+-
```
This does not hurt, but I don't see a reason to touch it either?

adschm commented on target/linux/layerscape/patches-5.4/302-dts-0074-arm64-dts-fsl-ls1028a-Add-Felix-switch-port-DT-node.patch:
```diff
@@ -48,55 +48,55 @@ Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
 +				reg = <0x000500 0 0 0 0>;
 +
 +				ports {
-+					#address-cells = <1>;
```
Any reason for all those indent-changes? Same in the following patch?

Am I not seeing something here?

adschm commented:
Your layerscape indent changes are really strange. Do you have something editing these files in the background?

graysky2 commented:
I use vimdiff.  My `~/.vimrc`:

```
"set autoindent
set tabstop=2
set shiftwidth=2
set smartindent

syntax on
filetype indent on
set t_Co=256
set background=dark
set nopaste
set pastetoggle=<F2>
map <F5> :setlocal spell! spelllang=en_us<CR>
highlight Comment cterm=italic
set showmatch     " show matching brackets (),{},[]
set mat=5         " show matching brackets for 0.5 seconds

"http://stackoverflow.com/questions/4998582/show-whitespace-characters-in-gvim
map <F4> :set list!<CR>
set listchars=eol:$,tab:>-,trail:~,extends:>,precedes:<
highlight SpecialKey term=standout ctermbg=yellow guibg=yellow
highlight RedundantSpaces term=standout ctermbg=Grey guibg=#ffddcc

set mouse-=a
```

graysky2 commented:
I am happy for you to make the adjustments you called out above.  I am uncomfortable editing the patches since they are diffs of diffs.  My SOP is to make these edits on the files and have quilt fix them up.  I am not sure how to do this without repeating the entire process again which will be laborious give the number times the patch target has been affected.

adschm commented:
Probably it will be quicker to start from scratch again that revert all those strange indent changes on _added_ lines ...

graysky2 commented:
@adschm - When I viewed them initially, I saw no strange indents.  Can you point me to a vimrc that sets up the indents correctly?

adschm commented:
> @adschm - When I viewed them initially, I saw no strange indents. Can you point me to a vimrc that sets up the indents correctly?

I have no idea what happened on your side. It would have simply been enough for the editor to not _change_ any indents. For me, most of the manually changed patches in your list didn't require any change at all (your initial indent change seemed to have a long tail of resulting adjustments ...)

graysky2 commented:
That is disturbing.  This is the same setup I have been using forever.  In your hands, how many patches required modification?  You mentioned most did not.

graysky2 commented:
@adschm - I rebased after fixing that.  Please let me know.

adschm commented on target/linux/layerscape/patches-5.4/302-dts-0074-arm64-dts-fsl-ls1028a-Add-Felix-switch-port-DT-node.patch:
```diff
@@ -58,7 +58,7 @@ Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
 +					#address-cells = <1>;
 +					#size-cells = <0>;
-+
++				
```
You still add whitespaces into an empty line here without any (apparant) reason ...

adschm commented on target/linux/layerscape/patches-5.4/302-dts-0112-arm64-dts-fsl-ls1028a-prepare-dts-for-overlay.patch:
```diff
@@ -400,39 +400,35 @@ Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
  					#address-cells = <1>;
  					#size-cells = <0>;
- 
+ 				
```
Again, whitespaces.

adschm commented on target/linux/layerscape/patches-5.4/302-dts-0112-arm64-dts-fsl-ls1028a-prepare-dts-for-overlay.patch:
```diff
@@ -400,39 +400,35 @@ Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
 +						status = "disabled";
  					};
 -					switch_port1: port@1 {
-+
```
No reason to remove these separating empty lines?

adschm commented:
Better now, but not yet resolved.

My suspicion is that you had $something automagically messing with indent/empty lines on the first layerscape file you touched. All other manual adjustments were then results of the mess you created in the first file (where the same mechanism seemed to have done additional harm in those). IIRC, I only touched the generic patch and one in layerscape manually; but I only did a quick refresh on late rc state for that bump, so I haven't documented.

adschm commented:
I've started another run myself. Will push it somewhere in 2-3 hours when it's done.

adschm commented:
https://github.com/adschm/openwrt/commits/k54new

graysky2 commented:
It's different from mine in a few key spots.

adschm commented:
Just deleted the one and manually touched the second pending and one layerscape patch.

graysky2 commented:
Shall we close this PR and just use yours once tested?

adschm commented:
You may take mine, test it, and push it here. Just put your message there, I will add/paste SoB tomorrow. I do not really have time for or interest in testing myself ATM.

graysky2 committed:
kernel: bump 5.4 to 5.4.86

Removed upstreamed patches:
  generic/pending-5.4/499-mtd-parser-cmdline-Fix-parsing-of-part-names-with-co.patch

Manually merged:
  adschm to edit

Build system: x86_64
Build-tested: ipq806x/R7800, bcm27xx/bcm2711
Run-tested: ipq806x/R7800

No dmesg regressions, everything functional

Signed-off-by: John Audia <graysky@archlinux.us>

Borromini commented:
I've build and run-tested this, so if the indentation stuff was the only issue then it looks good.

Build-tested: ath79/{generic,tiny}, ipq40xx, octeon, ramips/mt7621, realtek, x86/64.
Run-tested: realtek

Tested-by: Stijn Segers <foss@volatilesystems.org>

graysky2 commented:
> You may take mine, test it, and push it here. Just put your message there, I will add/paste SoB tomorrow. I do not really have time for or interest in testing myself ATM.

Done

adschm commented:
You still have whitespace issues in the latest version you pushed, no idea why. But since we have enough testing now, I've simply pushed my version ...

graysky2 commented:
>You still have whitespace issues in the latest version you pushed, no idea why. But since we have enough testing now, I've simply pushed my version ...

Downloaded your commit and applied it to my branch

```
wget https://github.com/adschm/openwrt/commit/4c6766cc17ec84b29e9d0655f5a6ea19f647d251.patch
```

adschm commented:
But you _do_ see the leading tabs added in your version (e.g. layerscape/patches-5.4/302-dts-0074-arm64-dts-fsl-ls1028a-Add-Felix-switch-port-DT-node.patch) that are not there in mine?

graysky2 commented:
Here is a screenshot of `target/linux/layerscape/patches-5.4/302-dts-0074-arm64-dts-fsl-ls1028a-Add-Felix-switch-port-DT-node.patch` in my editor.  Below it is the same screenshot but with whitespace characters highlighted.  How does this look in comparison to in your editor?

![1](https://user-images.githubusercontent.com/534822/103445103-0b86d080-4c3e-11eb-8997-c0596c915663.png)
![2](https://user-images.githubusercontent.com/534822/103445105-0de92a80-4c3e-11eb-9586-35d882260b83.png)
